include/SFML/System/Clock.hpp

```diff
@@ -74,11 +123,28 @@ class SFML_SYSTEM_API Clock
     Time restart();
 
 private:
+    using ClockImpl = typename priv::MostSuitableClockDetector::type;
+
+    static_assert(ClockImpl::is_steady,
+        "Provided implementation is not a monotonic clock");
+    static_assert(std::ratio_less_equal_v<ClockImpl::period, std::micro>,
+        "Clock resolution is too low. Expecting at least a microsecond precision");
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Convert clock duration to microseconds
+    ///
+    /// This function acts as a utility for converting clock
+    /// duration type instance into sf::Time
+    ///
+    /// \return Time in microseconds
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] static Time durationToTime(ClockImpl::duration duration);
```

Comment:
(Bikeshed) I don't think this is a good `[[nodiscard]]` candidate (I think I missed/misjudged this before).

Yes, the return value must be used, but so do 95% of all functions which return something. We don't want that attribute on functions like `Sprite::getPosition()` or `Sound::isRelativeToListener()`, even if they'd _technically_ qualify. This would litter our APIs without real-world benefit, and drown actual cases where `[[nodiscard]]` has a deeper meaning.

`[[nodiscard]]` is primarily useful to prevent mistakes in non-obvious cases where a return value could be accidentally ignored. For example:
```cpp
sf::Image image;
image.loadFromFile(...);
// WARN: didn't check 'bool' return type!
```
Same for methods which could be interpreted in a different way due to English language ambiguity:
```cpp
std::vector<T> container = ...;

container.empty(); // empty the container
// WARN: this is a noun, not a verb!
```

Here, we have a conversion function with a name `durationToTime()` -- it's 100% obvious that we get a `duration` and return a `Time`.

Reply:
Exactly, that was my thinking but I changed it anyway.

I still would like to push this further and completely propose the replacement of `sf::Time` with Chrono as discussed [here](https://en.sfml-dev.org/forums/index.php?topic=21639.15).  Then `durationToTime` would be removed.